src/planwise/client/projects2/components/settings.cljs

```diff
@@ -125,92 +145,134 @@
+                                       :disabled?          read-only}]])
+
+(defn- current-project-step-actions
+  [read-only current-project build-actions upgrade-actions]
+
+  [:section {:class-name "project-settings-section"}
+   [section-header 5 "Actions"]
+   [:div [:p [m/Icon "account_balance"] "Available budget"]]
+   [current-project-input "" [:config :actions :budget] "number" "$" "" {:disabled read-only :class "project-setting"}]
+   [m/TextFieldHelperText {:persistent true} "Planwise will keep explored scenarios below this maximum budget"]
+
+   [:div [:p [m/Icon "domain"] "Building a new provider..."]]
+   [listing-actions {:read-only?  read-only
+                     :action-name :build
+                     :list        build-actions}]
+
+   [:div [:p [m/Icon "arrow_upward"] "Upgrading a provider so that it can satisfy demand would cost..."]]
+   [current-project-input "" [:config :actions :upgrade-budget] "number" "$" "" {:disabled read-only :class "project-setting"}]
+
+   [:div [:p [m/Icon "add"] "Increase the capactiy of a provider by..."]]
+   [listing-actions {:read-only?   read-only
+                     :action-name :upgrade
+                     :list        upgrade-actions}]])
+
+(defn- current-project-step-review
+  [read-only current-project]
+  [:section {:class "project-settings-section"}
+   [section-header 6 "Review"]
+   [:div {:class "step-info"} "After this step the system will search for different improvements scenarios based on the given parameters. Once started, the process will continue even if you leave the site. From the dashboard you will be able to see the scenarios found so far, pause the search and review the performed work."]
+   [:div [:p [m/Icon "location_on"] "Kenya health facilities - ResMap 8902"]]
+   [:div [:p [m/Icon "account_balance"] "K 25,000,000"]]
+   [:div [:p [m/Icon "people"] "Kenya census 2005"]]
+   [:div [:p [m/Icon "directions"] "120 min walking distance, 40 min driving"]]
+   [:div [:p [m/Icon "info"] "A hospital with a capacity of 100 beds will provide service for 1000 pregnancies per year"]]])
+
 (defn current-project-settings-view
-  [{:keys [read-only]}]
+  [{:keys [read-only step]}]
   (let [current-project (subscribe [:projects2/current-project])
         build-actions   (subscribe [:projects2/build-actions])
         upgrade-actions (subscribe [:projects2/upgrade-actions])
         tags            (subscribe [:projects2/tags])]
-    (fn [{:keys [read-only]}]
-      [m/Grid {}
+    (fn [{:keys [read-only step]}]
+      [m/Grid {:class-name "wizard"}
+       [m/GridCell {:span 12 :class-name "steps"}
+        (doall
+         (map-indexed (fn [i iteration-step]
+                        [:a {:key i
+                             :class-name (join " " [(if (= iteration-step step) "active") (if (s/valid? (keyword (str "planwise.model.project-" iteration-step) "validation") @current-project) "complete")])
```

Comment:
Dynamically building the spec name at runtime is discouraged. A better approach would be to have a map from step to spec. Maybe even you can put more data in that map, like section title, link to next section, etc.

Reply:
On second read, you might want to have a vector which includes all this info. Eg.

```
(let [sections [{:step :goal :spec :planwise.model.project/goal :title "Goal"}
                {:step :consumers :spec :planwise.model.project/consumers :title "Consumers"}
                ...]]
  (map-indexed
    (fn [i section]
      [:a {:key section :class-name "..."} [:div (:title section)]]))
```